### PR TITLE
Change objects.h to get better code quality on Xtensa.

### PR DESCRIPTION
**Pull request by erikcorry**
Old code:

400e7de8 <toit::Object::byte_content(toit::Program*, unsigned char const**, int*, toit::BlobKind) const>: ...
400e7dfc:	ffcb22        	addi	a2, a11, -1
400e7dff:	002222        	l32i	a2, a2, 0
...
400e7e24:	2b3b      	addi.n	a2, a11, 3
400e7e26:	0238      	l32i.n	a3, a2, 0
400e7e28:	0183d6        	bgez	a3, 400e7e44 <toit::Object::byte_content(toit::Program*, unsigned char const**, int*, toit::BlobKind) const+0x5c>
400e7e2b:	2bbb      	addi.n	a2, a11, 11
400e7e2d:	0268      	l32i.n	a6, a2, 0
400e7e2f:	0e2d      	mov.n	a2, a14
400e7e31:	fc5656        	bnez	a6, 400e7dfa <toit::Object::byte_content(toit::Program*, unsigned char const**, int*, toit::BlobKind) const+0x12>
400e7e34:	bb7b      	addi.n	a11, a11, 7
400e7e36:	f27c      	movi.n	a2, -1
400e7e38:	0bb8      	l32i.n	a11, a11, 0

New code:

400e7764 <toit::Object::byte_content(toit::Program*, unsigned char const**, int*, toit::BlobKind) const>: ...
400e7776:	ffc292        	addi	a9, a2, -1
400e7779:	002982        	l32i	a8, a9, 0
...
400e77a0:	1938      	l32i.n	a3, a9, 4
400e77a2:	0123d6        	bgez	a3, 400e77b8 <toit::Object::byte_content(toit::Program*, unsigned char const**, int*, toit::BlobKind) const+0x54>
400e77a5:	3968      	l32i.n	a6, a9, 12
400e77a7:	0e2d      	mov.n	a2, a14
400e77a9:	fc7656        	bnez	a6, 400e7774 <toit::Object::byte_content(toit::Program*, unsigned char const**, int*, toit::BlobKind) const+0x10>
400e77ac:	f67c      	movi.n	a6, -1
400e77ae:	2928      	l32i.n	a2, a9, 8

### CODE REVIEW

File: src/objects.h [hunk 1]
```diff
@@ -368,19 +368,21 @@ class HeapObject : public Object {
 
   void _set_header(Program* program, Smi* id);
 
-  uword _raw() const { return reinterpret_cast<uword>(this) - HEAP_TAG; }
-  uword* _raw_at(int offset) { return reinterpret_cast<uword*>(_raw() + offset); }
-  const uword* _raw_at(int offset) const { return reinterpret_cast<const uword*>(_raw() + offset); }
+  inline uword _raw() const { return reinterpret_cast<uword>(this) - HEAP_TAG; }
+  inline uword* _raw_at()           { return reinterpret_cast<uword*>(reinterpret_cast<uword>(this) - HEAP_TAG); }
```
Comment (kasperl) [inline src/objects.h:372]:
```suggestion
  inline uword* _raw_at() { return reinterpret_cast<uword*>(reinterpret_cast<uword>(this) - HEAP_TAG); }
```

File: src/objects.h [hunk 2]
```diff
@@ -368,19 +368,21 @@ class HeapObject : public Object {
...
+  inline uword* _raw_at(int offset) { return reinterpret_cast<uword*>(reinterpret_cast<uword>(this) - HEAP_TAG + offset); }
+  inline const uword* _raw_at()           const { return reinterpret_cast<const uword*>(reinterpret_cast<uword>(this) - HEAP_TAG); }
```
Comment (kasperl) [inline src/objects.h:374]:
```suggestion
  inline const uword* _raw_at() const { return reinterpret_cast<const uword*>(reinterpret_cast<uword>(this) - HEAP_TAG); }
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Comment by erikcorry** [comment]
$ cp build/esp32/toit.bin bad.bin
$ ls -al bad.bin good.bin
-rw-r--r-- 1 erik erik 1283808 Apr 29 12:01 bad.bin
-rw-r--r-- 1 erik erik 1281504 Apr 29 12:00 good.bin

**Pull request merged**